src/test/java/core/basesyntax/service/impl/BatchLoaderImplTest.java

```diff
@@ -0,0 +1,106 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoCsvImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitCrate;
+import core.basesyntax.service.BatchLoader;
+import core.basesyntax.service.activitiy.ActivityHandler;
+import core.basesyntax.service.activitiy.ActivityType;
+import core.basesyntax.service.activitiy.AddingHandler;
+import core.basesyntax.service.activitiy.RemovingHandler;
+import core.basesyntax.strategy.ActivitiesStrategy;
+import core.basesyntax.strategy.ActivitiesStrategyImpl;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class BatchLoaderImplTest {
+    private static BatchLoader batchLoader;
+    private static StorageDao storageDao;
+    private static ActivitiesStrategy activitiesStrategy;
+    private static Map<String, ActivityHandler> activityHandlerMap;
+    private static List<String> fileData;
+    private static List<core.basesyntax.model.FruitCrate> shopStorage;
+
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    @BeforeClass
+    public static void beforeClass() {
+        activityHandlerMap = new HashMap<>();
+        activityHandlerMap.put(ActivityType.BALANCE.getLetter(), new AddingHandler());
+        activityHandlerMap.put(ActivityType.SUPPLY.getLetter(), new AddingHandler());
+        activityHandlerMap.put(ActivityType.RETURN.getLetter(), new AddingHandler());
+        activityHandlerMap.put(ActivityType.PURCHASE.getLetter(), new RemovingHandler());
+        activitiesStrategy = new ActivitiesStrategyImpl(activityHandlerMap);
+        storageDao = new StorageDaoCsvImpl();
+        batchLoader = new BatchLoaderImpl(storageDao, activitiesStrategy);
+        fileData = new ArrayList<>();
+        shopStorage = new ArrayList<>(List.of(new FruitCrate("banana", 125),
+                new FruitCrate("apple", 83)));
+    }
+
+    @Before
+    public void setUp() {
+        fileData.clear();
+        fileData.addAll(List.of("type,fruit,quantity",
```

Comment:
You can simply write fileData = List.of(...

Reply:
Then I will get UnsupportedOperationException